src/command/template/templates.rs

```diff
@@ -1,124 +1,96 @@
+    });
+    let resp: ResponseData = request(&query)?;
+    Ok(resp.template)
+}
 
-    /// Get a template by ID
-    pub fn get(self, template_id: &str) -> Result<GithubTemplate> {
-        self.templates
-            .into_iter()
-            .find(|template| template.id == template_id)
-            .ok_or_else(|| {
-                let mut err = RoverError::new(anyhow!("No template found with id {}", template_id));
-                err.set_suggestion(Suggestion::Adhoc(
-                    "Run `rover template list` to see all available templates.".to_string(),
-                ));
-                err
-            })
-    }
+pub fn get_templates_for_language(
+    language: ProjectLanguage,
+) -> Result<Vec<GetTemplatesForLanguageTemplates>> {
+    use super::queries::get_templates_for_language::*;
+    let query = GetTemplatesForLanguage::build_query(Variables {
+        language: Some(language.into()),
+    });
+    let resp: ResponseData = request(&query)?;
+    error_if_empty(resp.templates)
+}
 
-    /// Filter templates by language
-    #[must_use]
-    pub fn filter_language(mut self, language: ProjectLanguage) -> Self {
-        self.templates = language.filter(self.templates);
-        self
-    }
+pub fn list_templates(
+    language: Option<ProjectLanguage>,
+) -> Result<Vec<ListTemplatesForLanguageTemplates>> {
+    use super::queries::list_templates_for_language::*;
+    let query = ListTemplatesForLanguage::build_query(Variables {
+        language: language.map(Into::into),
+    });
+    let resp: ResponseData = request(&query)?;
+    error_if_empty(resp.templates)
+}
 
-    /// Consume self and return the list of templates that were selected.
-    ///
-    /// # Errors
-    ///
-    /// Returns an error if there were no matching templates.
-    pub fn values(self) -> Result<Vec<GithubTemplate>> {
-        if self.templates.is_empty() {
-            Err(RoverError::new(anyhow!(
-                "No templates matched the provided filters"
-            )))
-        } else {
-            Ok(self.templates)
-        }
+pub fn error_if_empty<T>(values: Vec<T>) -> Result<Vec<T>> {
+    if values.is_empty() {
+        Err(RoverError::new(anyhow!(
+            "No templates matched the provided filters"
+        )))
+    } else {
+        Ok(values)
     }
+}
 
-    /// Prompt to select a template
-    pub fn selection_prompt(self) -> Result<GithubTemplate> {
-        let mut templates = self.values()?;
-        let selection = Select::new()
-            .with_prompt("Which template would you like to use?")
-            .items(&templates)
-            .default(0)
-            .interact_on_opt(&Term::stderr())?;
+/// Prompt to select a template
+pub fn selection_prompt(
+    mut templates: Vec<GetTemplatesForLanguageTemplates>,
+) -> Result<GetTemplatesForLanguageTemplates> {
+    let names = templates
+        .iter()
+        .map(|t| t.name.as_str())
+        .collect::<Vec<_>>();
+    let selection = Select::new()
```

Comment:
we'll want to make sure the terminal is a tty before prompting on it so we don't hang in places like CI - if it's not a TTY we should error early and instruct them to pass required arguments instead (there are examples of this in `rover dev` code).

Reply:
Opened #1395 to address this in a follow-up, since it's not a change from the current released behavior.